### PR TITLE
Feat/toast composition hook

### PR DESCRIPTION
**Pull request by m0ksem**
closes https://github.com/epicmaxco/vuestic-ui/issues/1436

Now `VaToast` is independent of Vue app context and can be used in any method.

Added `useToast` hook and test demo.

## Example
```ts
const { init, close, closeAllCreatedInThisHook, closeAll } = useToast()
```

## Breaking Changes

`closeAllCreatedInThisHook` - used if user want to close all toasts that actually created in SetupContext. So if multiple toasts shown from different components (I mean setup context) it would close toasts from other components. Can be useful. Can you help with method name, or you find it ok?

`closeAll` by default closes toasts only created in app, where hook is used. It accepts boolean, if it true it will close toasts for any vue app created on page. I'm not sure about default behavior, if you find it good - let's keep it. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

**Review by RVitaly1978 — APPROVED** [review]

File: packages/ui/src/components/va-toast/VaToast.demo.vue [hunk 1]
```diff
@@ -34,22 +34,63 @@
...
     const { setGlobalConfig } = useGlobalConfig()
-
     setGlobalConfig(config => ({
       ...config,
       VaToast: {
         color: 'red',
       },
     }))
+
+    const { init, close, closeAllCreatedInThisHook, closeAll } = useToast()
```
Comment (LighthouseKeeperYN) [inline packages/ui/src/components/va-toast/VaToast.demo.vue:57]: I think we should either change variable names or never destructure toast object. Standalone "init" function looks too vague.

File: packages/ui/src/components/va-toast/VaToast.demo.vue [hunk 2]
```diff
@@ -34,22 +34,63 @@
...
+
+          init({ message: 'Now we try to close all toasts in 3 seconds' })
+          init({ message: 'Now we try to close all toasts in 3 seconds' })
+          init({ message: 'Now we try to close all toasts in 3 seconds' })
+          init({ message: 'Now we try to close all toasts in 3 seconds' })
+          init({ message: 'Now we try to close all toasts in 3 seconds' })
+
+          setTimeout(() => {
+            closeAll()
+          }, 3000)
```
Comment (LighthouseKeeperYN) [inline packages/ui/src/components/va-toast/VaToast.demo.vue:86]:
```suggestion
          setTimeout(closeAll, 3000)
```

File: packages/ui/src/components/va-toast/useToast.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  /**
+   * Closes all VaToast instances in current App instance.
+   * @param allApps if you have multiple vue apps on page, set allApps to true to close it for all apps.
+   */
+  const closeAll = (allApps = false) => closeAllNotifications(allApps ? undefined : appContext)
+
+  /** Use this method if you need to close only toasts that created with this hook */
+  const closeAllCreatedInThisHook = () => {
+    createdInThisSetupContext.forEach((id) => closeById(id))
```
Comment (LighthouseKeeperYN) [inline packages/ui/src/components/va-toast/useToast.ts:31]:
```suggestion
    createdInThisSetupContext.forEach(closeById)
```

**Review by LighthouseKeeperYN — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/va-toast/VaToast.demo.vue (hunk 1 above)
Comment (m0ksem) [reply packages/ui/src/components/va-toast/VaToast.demo.vue:57]: User can change it's name
```ts
const { init: initToast, close, closeAllCreatedInThisHook, closeAll } = useToast()
```
or simply

```ts
const toast = useToast()
```


I just left it as it was in options API `$vaToast.init()`

File: packages/ui/src/components/va-toast/useToast.ts (hunk 3 above)
Comment (m0ksem) [reply packages/ui/src/components/va-toast/useToast.ts:31]: Actually, passing id argument is much readable. Also, if we will add some additional argument to closeById method, this forEach can cause some bugs. I would prefer to use a new arrow function as callback.

**Review by LighthouseKeeperYN — APPROVED** [review]

**Comment by asvae** [comment]
Thanks!
Probably needs some docs as well, so that users will be able to use it.

**Comment by asvae** [comment]
> if it true it will close toasts for any vue app created on page
Interesting approach. Could be useful is someone attempts [micro frontends](https://micro-frontends.org/). We might want to enforce this in architecture notes.

**Comment by m0ksem** [comment]
My bad. I've forgotten about docs.

**Comment by m0ksem** [comment]
I added some docs here.
image
image

**Pull request merged**